src/utils.jl

```diff
@@ -0,0 +1,115 @@
+using HTTP
+using JSON3
+using Dates
+
+function prepare_cohort_directory()
+    path = joinpath(@__DIR__, "..", "data", "cohort_definitions")
+    mkpath(path)
+    return path
+end
+
+function load_metadata(
+    metadata_path::String,
+    metadata_check::Bool
+)
+    if metadata_check && isfile(metadata_path)
+        return JSON3.read(read(metadata_path, String), Dict{String, Any})
+    end
+    return Dict{String, Any}()
+end
+
+function write_metadata_entry(
+    metadata::Dict{String, Any},
+    id::Int,
+    version::Int,
+    last_modified::String
+)
+    metadata[string(id)] = Dict(
+        "id" => id,
+        "version" => version,
+        "lastModified" => last_modified,
+        "downloadedAt" => string(Dates.now())
+    )
+end
+
+function save_metadata(
+    metadata::Dict{String, Any},
+    metadata_path::String
+)
+    open(metadata_path, "w") do file
+        JSON3.pretty(file, metadata, JSON3.AlignmentContext(indent=4))
+    end
+end
+
+function save_cohort_json(
+    id::Int,
+    body,
+    save_dir::String
+)
+    path = joinpath(save_dir, "$(id).json")
+    open(path, "w") do file
+        JSON3.pretty(file, body, JSON3.AlignmentContext(indent=4))
+    end
+    return path
+end
+
+function format_timestamp(
+    ms::Int
+)
+    date = Dates.unix2datetime(ms ÷ 1000)
+    return Dates.format(date, "yyyy-mm-ddTHH:MM:SS")
+end
+
+function get_cohort_definition(
+    IDs;
+    progress_bar::Bool = true,
+    metadata_check::Bool = true
+)
+    save_dir = prepare_cohort_directory()
```

Comment:
Oh I think I may have confused you in our earlier discussions! With this, please make the `save_dir` a kwarg that the user can define. Otherwise, the `save_dir` should just be the user's current working directory.